### PR TITLE
First set of plugin docs

### PR DESCRIPTION
**Pull request by seancdavis**
The main features of these changes include:

- New docs components
- New plugin docs pages
- Concepts reference

## New Docs Components

As part of this work, there are several new components for working with docs (in general) and (more specifically) plugin docs. 

## New Docs

This work includes the following new pages:

- /docs/plugins — This is the landing page of the Plugins docs section. I pulled it from /docs/installation/plugins and replaced the installation page with a quick installation example and then guide the user over to this new page.
- /docs/plugins/grouparoo-intercom
- /docs/plugins/grouparoo-postgres
- /docs/plugins/grouparoo-sqlite

While I really really really don't like MDX, nor do I think it appropriate to put content in React components, sharing content in this way seemed like the best way to not duplicate content, while providing a means for taking advantage of pulling values from elsewhere.

For example, there is now the option to add a `configOptions` object on the objects within the `data/plugins.ts`. These then get rendered in the installation section for each plugin.

## Concepts Reference

I reworked /docs/getting-started/product-concepts to be our dictionary, so to speak. It is now a collection of base-level terms that we can link to from anywhere else. It's not a great solution, but it's a place to start.

## Other Changes

There are also a handful of updates to other docs here. Most of those were reviewed in the first pass.

## Next Steps

The proposal is that we merge this content in and then build stories for each individual docs page that we want to add. It's a bit messy, but at this point we just need to get the content out there.

### CODE REVIEW

**Comment by seancdavis** [comment]
### Review v1

The [Postgres plugin doc](https://www-grouparoo-com-git-177009144-plugin-docs-grouparoo.vercel.app/docs/plugins/grouparoo-postgres) is ready for a review. There are a few questions and items to note while looking over this page.

Here are some general notes:

- I've also updated the [Sources](https://www-grouparoo-com-git-177009144-plugin-docs-grouparoo.vercel.app/docs/config/sources/community), [Properties](https://www-grouparoo-com-git-177009144-plugin-docs-grouparoo.vercel.app/docs/config/properties/community), and [Destinations](https://www-grouparoo-com-git-177009144-plugin-docs-grouparoo.vercel.app/docs/config/destinations/community) docs to include what I'm calling _common options_ — those options that are shared across all config objects of that type.
- The Postgres plugin seems like a great place to start, as it is both a Source and a Destination. After doing a few of these, I'll start to get a sense of where we can abstract and share sections of repeated content. But any help or advice you can offer in that area to help streamline the process would be super duper.

And here are the questions/concerns I have:

1. I don't think I nailed the common vs unique options in every case here. A lot of it was a guess.
2. Why are `mapping` and `filters` outside of `options`? Are they common config options or is there some other reason?
3. Would it be beneficial to have a columnar/table page to store column-based source information just one time? Or is it good to start here, being more individualistic, and then see where that takes us?
4. Any ideas for the in-page nav? It's getting a little messy. But I named my headings intentionally to help with SEO.

File: components/docs/pluginsList.tsx [hunk 1]
```diff
@@ -33,9 +34,16 @@ export default function PluginsList() {
...
+        .replace(/\@/g, "") // Characters to remove
+        .replace(/[\ \/]/g, "-"); // Characters to replace with a hyphen
+      return `/docs/plugins/${name}`;
+    };
+
     const packages: NPMPackage[] = response.objects
       .filter((o) => o.package)
-      .map((o) => o.package)
+      .map((o) => ({ ...o.package, url: pluginDocsUrl(o.package) }))
       .filter((o) => !ignoredPlugins.includes(o.name))
```
Comment (evantahler) [inline components/docs/pluginsList.tsx:62]: We may want to hide some plugins too, like perhaps `app-templates` doesn't belong

File: pages/docs/config/destinations/community.mdx [hunk 2]
```diff
@@ -79,15 +79,59 @@ This will give you something like the following:
...
+    LNAME: "lastName",
+  }
+}
+```
+
+Notice the odd structure of the keys. That's specific to Mailchimp. Each Destination's config file will be generated will some helper comments to get you started.
+
+### Destination Group Membership
+
+Grouparoo also provides the ability to attach group membership to the exported data. How this looks within the Destination is specific to the Destination. But the shape of the config file largely remains the same.
```
Comment (evantahler) [inline pages/docs/config/destinations/community.mdx]: ... In most Destinations, Destinations Group Memberships become either tags (Mailchimp), Lists (Hubspot), or appended as additional properties to the Profile (CustomerIO)

File: pages/docs/config/properties/community.mdx [hunk 3]
```diff
@@ -16,19 +16,17 @@ Now that you have a Source or some Sources ([see here if you don't have a Source
 
 ## Generating a New Property
 
-You are likely to have more Profile Properties than any other type of config object in your Grouparoo application. Because of this, creating them with the CLI can be a tedious venture. While we still recommend using the CLI, we've outlined two approaches to take when creating Properties.
-
-### Method #1: Using the CLI
+You are likely to have more Profile Properties than any other type of config object in your Grouparoo application. Because of this, creating them with the CLI can be a tedious venture. While we still recommend using the CLI, you can also generate a single Property and the duplicate the code inside one file or among multiple files.
```
Comment (evantahler) [inline pages/docs/config/properties/community.mdx]: > You are likely to have more ~~Profile~~ Properties

(Profile Properties are the instance of the Property on the Profile)

File: pages/docs/config/properties/community.mdx [hunk 4]
```diff
@@ -16,19 +16,17 @@ Now that you have a Source or some Sources ([see here if you don't have a Source
...
 
 To generate a new Property for your Grouparoo application, you can run [the `generate` command](/docs/cli/config#generate).
 
 For example, let's say you have configured a Postgres Table Source with the ID `users`. And let's say you want to add a Property that calculates the lifetime value of each customer. You can generate a new Profile for that Source like this:
 
     grouparoo generate postgres:table:property ltv --parent users_table
 
-Note here that `ltv` is the ID for the Source. An ID is always required when generating a config object.
+Note here that `ltv` is the `id` for the Property. An `id` is always required when generating a config object.
```
Comment (evantahler) [inline pages/docs/config/properties/community.mdx:27]: Good catch!

File: pages/docs/config/sources/community.mdx [hunk 5]
```diff
@@ -72,19 +72,55 @@ This will give you something like the following:
...
+- `type`: (**required**) The type of Source will vary based on the generator you're using. You likely don't want to change this value.
+- `appId`: (**required**) The `id` of the App (connection) that the Source uses.
+
+The generator makes a _reasonable_ guess at the values in the config file it generated (in the `config/sources` directory). There are ellipses (`"..."`) in the places in which it couldn't make a reasonable guess. In the example above, after generating a Postgres Source, all but `appId` and `options.table` were pre-populated.
 
 Other sections may be commented out but require some action by you. The best practice when configuring an Source is to read the comments, keys, and values within the generated file and fill in the appropriate values for your Source.
 
+### Bootstrapped Properties
+
+The very first Source you create in Grouparoo requires a Bootstrapped Property. Think of this as the primary key for all Profiles. Source config files provide you the ability to define this Bootstrapped Property. It will look like a simplified version of [a Property](/docs/config/properties/community).
```
Comment (evantahler) [inline pages/docs/config/sources/community.mdx]: ... This Bootstrapped Property must be unique to all of your Profiles, and is commonly a User ID or email address.

File: pages/docs/plugins/grouparoo-postgres.mdx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+Mapping is a section of key-value pairs which represents how Grouparoo should match source columns to properties in Grouparoo.
+
+For example, let's say your database has a column named `email` and `first_name`, while the Property `id` values in Grouparoo are `emailAddress` and `firstName`. Your mapping would look like this:
+
+```js
+mapping: {
+  email: "emailAddress",
+  first_name: "firstName"
+}
+```
```
Comment (evantahler) [inline pages/docs/plugins/grouparoo-postgres.mdx]: You only map one column

File: pages/docs/plugins/grouparoo-postgres.mdx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+#### Table Source Schedule
+
+A schedule is attached to every Source config file by default, commented out. If you want to import data from the source on a schedule (which is the typical behavior), then you should remove the comments and apply the necessary values.
+
+[You can read more about the common options here](/docs/config/sources/community#adding-a-schedule). The Postgres-specific options (those in the `options` object) are:
+
+- `column`: (**required**) The name of the column to use as the high water mark. This is a date column.
```
Comment (evantahler) [inline pages/docs/plugins/grouparoo-postgres.mdx]: We should talk more about what a high water mark is:
* Increasing over time (updated_at or user_id)
* we will only import things that are newer than last time

File: pages/docs/plugins/grouparoo-postgres.mdx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+A Query Source is a more flexible way to build properties. With a Query Source, you can add custom SQL commands to your Properties, which could pull data from one or more tables in your database.
+
+The unique thing about the way a Query Source works is that there aren't any unique options for the Source itself. But the Schedule and Properties look a little different.
+
+#### Query Source Schedule
+
+Like a [Table Source Schedule](#table-source-schedule), a Query Source Schedule is included with the generated config file, commented. It has a couple unique options:
+
+- `query`: A SQL query to return that tells Grouparoo which Profiles to check each time the interval occurs.
+- `propertyId`: The `id` of the Grouparoo Property whose data is returned by `options.query`.
```
Comment (evantahler) [inline pages/docs/plugins/grouparoo-postgres.mdx]: ... be sure that your query only returns on column of data, and that it maps to the Grouparoo Property you set with `options.propertyId`.

File: pages/docs/plugins/grouparoo-postgres.mdx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+    $ grouparoo generate postgres:query:property lifetime_value --parent purchases
+
+This will drop a file at `config/properties/lifetime_value.js`. Edit this file to match your desired configuration.
+
+There is one unique option for Query Source Properties:
+
+- `query`: The query to extract the Property. You can use [mustache variables](https://github.com/janl/mustache.js#variables) to represent the keys of other Properties in the system.
+
+Here's an example that sums the values in the `price` column for rows in which the `user_id` column's value matches the value of the Grouparoo Profile's `userId` field (i.e. `userId` is the `id` for the Property in Grouparoo):
```
Comment (evantahler) [inline pages/docs/plugins/grouparoo-postgres.mdx]: You can use any of the other Properties you've defied in Grouparoo, and they are available to use within mustache `{{ }}` brackets.

File: pages/docs/plugins/grouparoo-postgres.mdx [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+This would generate a config file at `config/destinations/my_export.js` in your Grouparoo project. Edit this file to match your desired configuration.
+
+The Postgres Destination has a few unique options:
+
+- `table`: The name of the table which Grouparoo which use to export **Profiles**.
+- `primaryKey`: The primary key for the specified table.
+- `groupsTable`: Grouparoo also exports group memberships. This is the table used to export Group data.
+- `groupForeignKey`: The foreign key that the groups table uses to reference the profiles table.
+- `groupColumnName`: The column name used to store the Group name values.
+
```
Comment (evantahler) [inline pages/docs/plugins/grouparoo-postgres.mdx]: The Postgres Destination will make 2 tables - one for Profiles, and another normalized table for Group membership.

**Review by evantahler — COMMENTED** [review]
I think this is a really good start.   I'm curious in general to how many of these chunks can be re-used, and if some of this data can be sourced from the plugins (or the monorepo) itself to keep things up-to-date.

**Comment by seancdavis** [comment]
The PR description is updated with the current state of affairs here.

**Pull request merged**